assembly/src/assembler/mod.rs

```diff
@@ -493,15 +496,22 @@ impl Assembler {
                     )
                     .with_span(proc_alias.span());
 
-                    let proc_alias_root = self.resolve_target(
+                    let proc_mast_root = self.resolve_target(
                         InvokeKind::ProcRef,
                         &proc_alias.target().into(),
                         &pctx,
                         mast_forest_builder,
                     )?;
+
+                    // insert external node into the MAST forest for this procedure; if a procedure
+                    // with the same MAST rood had been previously added to the builder, this will
+                    // have no effect
+                    let proc_node_id = mast_forest_builder.ensure_external(proc_mast_root)?;
+                    let procedure = pctx.into_procedure(proc_mast_root, proc_node_id);
```

Comment:
`into_procedure()`'s docs specify:

> The passed-in `mast_root` defines the MAST root of the procedure's body while `mast_node_id` specifies the ID of the procedure's body node in the MAST forest in which the procedure is defined.

But here, `mast_node_id` refers to an external node. I think we should update the docs there to clarify that the "body" can also be an `ExternalNode`, which is what re-exported procedures get compiled down to.

Reply:
Agreed! Added more info to the description.